renormalizer/model/basis.py

```diff
@@ -382,45 +383,253 @@ def op_mat(self, op: Union[Op, str]):
         if op_symbol == "I":
             mat = np.eye(self.nbas)
         
-        elif op_symbol.split("^")[0] == "x":
+        elif op_symbol.split("^")[0] == "x" and " " not in op_symbol:
```

Comment:
Is these code just for checking or it does something different from the good old `mat = np.diag(self.dvr_x ** moment)`?

Reply:
The modification ensures analytical accuracy for the operators such as `x p`, `x^2`, `x^3`, `x p^2` etc.

Reply:
Am I to understood that even for `x` `mat = np.diag(self.dvr_x)` is not accurate?

Reply:
For `x`, it is exact. But for `x^2`, `x^3`, etc, it is not.